src/state/slices/marketDataSlice/marketDataSlice.tsx

```diff
@@ -163,25 +168,38 @@ export const marketApi = createApi({
       },
       keepUnusedDataFor: 5, // Invalidate cached asset market data after 5 seconds.
     }),
-    findPriceHistoryByAssetId: build.query<HistoryData[] | null, FindPriceHistoryByAssetIdArgs>({
-      queryFn: async (args, { dispatch }) => {
-        const { assetId, timeframe } = args
-        if (shouldIgnoreAsset(assetId)) return { data: [] }
-        try {
-          const data = await getMarketServiceManager().findPriceHistoryByAssetId({
-            timeframe,
-            assetId,
-          })
-          const payload = { args, data }
-          dispatch(marketData.actions.setCryptoPriceHistory(payload))
-          return { data }
-        } catch (e) {
-          const error = {
-            data: `findPriceHistoryByAssetId: error fetching price history for ${assetId}`,
-            status: 400,
-          }
-          return { error }
-        }
+    findPriceHistoryByAssetIds: build.query<null, FindPriceHistoryByAssetIdArgs>({
+      queryFn: async function findPriceHistoryByAssetIds(args, { dispatch }) {
+        const { assetIds, timeframe } = args
+
+        if (assetIds.length === 0) return { data: null }
+
+        const responseData: { assetId: AssetId; historyData: HistoryData[] }[] = await Promise.all(
+          assetIds.map(async assetId => {
+            if (shouldIgnoreAsset(assetId)) return { assetId, historyData: [] }
+            try {
+              const historyData = await getMarketServiceManager().findPriceHistoryByAssetId({
+                timeframe,
+                assetId,
+              })
+
+              return { assetId, historyData }
+            } catch (e) {
+              return { assetId, historyData: [] }
```

Comment:
We previously returned an error RTK query with `Error fetching price history`, I understand that we don't want to do this anymore if only *one* fails, but we probably don't want to swallow the error.
What about console.logging / console.erroring it, and re-assessing if this is too spewy in a follow-up?